lib/apiv2/clusterinfo.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiv2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	KindClusterInformation     = "ClusterInformation"
+	KindClusterInformationList = "ClusterInformationList"
+)
+
+// ClusterInformation contains the cluster specific information.
+type ClusterInformation struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the ClusterInformation.
+	Spec ClusterInformationSpec `json:"spec,omitempty"`
+}
+
+// ClusterInformationSpec contains the values of describing the cluster.
+type ClusterInformationSpec struct {
+	// ClusterGUID is the GUID of the cluster
+	ClusterGUID string `json:"clusterGUID,omitempty" validate:"omitempty"`
+	// ClusterType describes the type of the cluster
+	ClusterType string `json:"clusterType,omitempty" validate:"omitempty"`
+	// The version of the cluster
+	Version string `json:"version,omitempty" validate:"omitempty"`
```

Comment:
I think Version and CalicoVersion are the same thing really.  We can drop the Version one.

Reply:
Oh oops thought that it was the Cluster version.  I'll remove it